docs/resources/rules/assertions.md

```diff
@@ -0,0 +1,487 @@
+suggest | [`string`] | List of suggestions to display if the problem occurs.
+severity | `string` | The severity level of the problem if the assertion is false. It must be one of these values: `error`, `warn`, `off`. Default value is `error`.
+enum | [`string`] | Asserts value is within a predefined list of values. See [enum example](#enum-example).
+pattern | `string` | Asserts value matches a regex pattern. See [regex pattern example](#pattern-example).
+casing | `string` | Asserts a casing style from this possible list: `camelCase`, `kebab-case`, `snake_case`, `PascalCase`, `MACRO_CASE`, `COBOL-CASE`, `flatcase`. See [casing example](#casing-example).
+mutuallyExclusive | [`string`] | Asserts list of properties (key names only) are mutually exclusive. Use `$keys` in the `on` when this is enabled (for example, `Operation.$keys`). See [mutuallyExclusive example](#mutuallyexclusive-example).
+mutuallyRequired | [`string`] | Asserts list of properties (key names only) are mutually required. Use `$keys` in the `on` when this is enabled (for example, `Operation.$keys`). See [mutuallyRequired example](#mutuallyrequired-example).
+required | [`string`] | Asserts all listed values are defined. See [required example](#required-example).
+disallowed | [`string`] | Asserts all listed values are not defined. See [disallowed example](#disallowed-example).
+defined | `boolean` | Asserts a property is defined. See [defined example](#defined-example).
+undefined | `boolean` | Asserts a property is undefined. See [undefined example](#undefined-example).
+nonEmpty | `boolean` | Asserts a property is not empty. See [nonEmpty example](#nonempty-example).
+minLength | `integer` | Asserts a minimum length (inclusive) of a string or list (array). See [minLength example](#minlength-example).
+maxLength | `integer` | Asserts a maximum length (exclusive) of a string or list (array). See [maxLength example](#maxlength-example).
+sortOrder | `string` \| [Sort Order object](#sort-order-object) | Asserts a sort order to a list (array) of strings or a list of objects by one of the object's property values. See [sortOrder example](#sortorder-example).
+
+## Context object
+
+Property | Type | Description
+-- | -- | --
+type | `string` | **REQUIRED.** One of the [OpenAPI node types](#openapi-node-types).
+filterKeys | [`string`] | The list of key names to continue evaluation with respect to the subject.
+
+See the [context example](#context-example).
+
+## Sort Order object
+
+Property | Type | Description
+-- | -- | --
+direction | `string` | **REQUIRED.** Asserts values are sorted in ascending or descending order using one of these possible values: `asc` or `desc`.
+property | `string` | **REQUIRED.** The name of the property in the collection of objects that will hold values to assess the sort order.
+
+## Examples
+
+The following example shows two assertions each multiple asserts (`defined`, `minLength`, `maxLength`, `pattern`).
```

Comment:
Frankly I see four but not two assertions in example below, but I understand that technically there are only two :) I think you divided it because of message as it should be exact about the subject. We could combine it within `subject` key, like

```yaml
lint:
  rules:
    assertions:
      - subject: 
        - name: Tag
          message: Tag description must be at least 30 characters and end with a full stop.
        - name: Operation
          message: Operation description must be at least 30 characters and end with a full stop.
        - name: Info
          message: Info description must be at least 30 characters and end with a full stop.
        property: description
        severity: error
        defined: true
        minLength: 30
        pattern: /\.$/
```
If the message is about the property, we can put it under `property` key, for example
```yaml
lint:
  rules:
    assertions:
      - subject: Tag
        property: 
        - name: description
          message: Tag description must be at least 30 characters and end with a full stop.
        - name: name
          message: Tag name must be at least 30 characters and end with a full stop.
        severity: error
        defined: true
        minLength: 10
        pattern: /\.$/
```
If we want to use one message for all the asserts, so let's do it as previous at the assertion level.
So now we will have three ways to use `message` - under `subject`, under `property` and at the assertion level.

Reply:
Actually, you are right. I adjusted it to four assertions but didn't show it here. We are trying to limit nesting as much as possible. While I am okay with the idea of make subject and property lists, I think it may result in overcomplexity and documenting it would be more difficult than making four assertions.

Reply:
```suggestion
The following example shows four assertions each multiple asserts (`defined`, `minLength`, `maxLength`, `pattern`).
```

Reply:
I think this is okay for properties:  ` string | [string] ` but not something like ` string | [NameMessage Object]`. Flatter with more assertions is easier to use.